### PR TITLE
Refactor balance sheet

### LINKED ISSUE #3265: Add Condition (Creditor or Debtor) in Balance Sheet
In the Ohada accounting system, some references to the Accounts are only taken into account if they have a Creditor or Debtor balance, this issue proposes to add this option in the OHADA Report module and report.

### PR DESCRIPTION
**Pull request by lomamech**
- Added the credit_balance and debit_balance property to account for certain account references that have a debit or credit
balance
- (test) Update integration test
- Update Account Reference display account selected only by solde(Creditor
  or Debitor)
- Searches the balance of the accounts that must have a credit or debit
  balance and checks whether these accounts comply with the parameters
  credit_balance and debit_balance

- Modification of the report Balance sheet, recovery of data conditional
  reference and verification of data given respective references to the
  balance (Debiteur and Crediteur)

- Update references accounts integration test

closes #3265

### CODE REVIEW

**Comment by jniles** [comment]
Tests pass.  I'm working on data migration and am not familiar enough with the OHADA system to say if this is correct.

@mbayopanda can you take a look a this?

File: test/integration/accountReferences.js [hunk 1]
```diff
@@ -56,7 +61,7 @@ describe('(/accounts/references) Accounts References', () => {
       .then((res) => {
         helpers.api.listed(res, numAccountReference);
         console.log('res : ', res.body);
```
Comment (mbayopanda) [inline test/integration/accountReferences.js]: Could you remove this `console.log`

**Review by mbayopanda — COMMENTED** [review]
LGTM 👍 

@lomamech could you attach the sql script of account references with ohada balance sheet references

**Review by mbayopanda — APPROVED** [review]
LGTM :+1

bors r+

**Pull request merged**